### PR TITLE
Mi Flora support

### LINKED ISSUE #44: Add flower-care fonction definitively
Hi,

Your work is awesome, the mijia hygrothermograph scanning is very reactive.
I try to use "https://github.com/honkmaster/homebridge-mi-flower-care" as the same time, but it doesn't work properly : 
(result of debug)
[2019-3-14 18:50:45] [Sonde Mijia 1] Stopped scanning.
[2019-3-14 18:50:45] [Sonde Mijia 2] Stopped scanning.
[2019-3-14 18:50:45] [Sonde Mijia 3] Stopped scanning.
[2019-3-14 18:50:45] [Sonde Mijia 4] Stopped scanning.
[2019-3-14 18:50:47] [Sonde Mijia 1] Start scanning.
[2019-3-14 18:50:47] [Sonde Mijia 2] Start scanning.
[2019-3-14 18:50:47] [Sonde Mijia 3] Start scanning.
[2019-3-14 18:50:47] [Sonde Mijia 4] Start scanning.

Would you add the flower care fonctionnality to your plugin ? 
For saving battery life, i think it's important to add a scan  interval to the flower sensors. 
(if you can add a possibility to change dynamicly this interval with a mqtt request eg : miflower/intervalscan  -> 300 for 5 minutes )

Thanks a lot for your reply.

### PR DESCRIPTION
**Pull request by Kylmakalle**
Hello @hannseman. I would Like to introduce a full support for Mi Flora accessories.
Main purpose if this PR is to use one (non-conflicting with other) plugin for all devices using BLE broadcast advertisement packets. Closes #44 and https://github.com/honkmaster/homebridge-mi-flower-care/issues/4

Everybody can test this with terminal command in homebridge. (Make sure to have a config backup)
```bash
sudo -E -n npm install -g Kylmakalle/homebridge-mi-hygrothermograph
```

- Added new (backward-compatible) `type` property for accessory
- Exposed `Fertility` as Light Sensor, `Moisture` as Humidity Sensor and `Illuminance` as Light Sensor
- Added MQTT support for exposed values.
- Updated UI config scheme with dynamic support for both Hygrothermograph and Mi Flora accessory types.
- Fully unit-tested and linted code
- Tested IRL

## Accessories view
I've tested it with my plant and it looks pretty fine.
image

image

## Config UI
image

image

## Downsides
- Plant Humidity will affect whole room humidity in Home app (can be fixed with separate room)
- Plant Fertility is exposed as Light Sensor due to most proper value support, which also affects whole room Light level in Home app. Subject to change.
image

### CODE REVIEW

**Comment by hannseman** [comment]
@Kylmakalle cool thanks! 🤩 

As you've noticed I've been quite slow at responding to this PR, it's a bit hectic for me at the moment but probably calm down around next week. Then I'll give this a review! Thanks again.

**Comment by Kylmakalle** [comment]
Hey @hannseman, it's alright! I hope that you are keeping safe and well.

I've added some more tests to acquire maximum coverage hit rate while you're taking your time.

File: lib/accessory.js [hunk 1]
```diff
@@ -60,6 +87,86 @@ class HygrothermographAccessory {
...
+  setFertility(newValue, force = false) {
+    if (newValue == null) {
+      return;
+    }
+    this.latestFertility = newValue;
+    this.lastUpdatedAt = Date.now();
+    if (this.useBatchUpdating && force === false) {
+      return;
+    }
+    // TODO: Fertility Accessory and Characteristic
```
Comment (hannseman) [inline lib/accessory.js:155]: I guess this is fine. We should probably add a small note in the README about the fertility being exposed as a light sensor to try and avoid getting issues about it :-)

**Review by hannseman — COMMENTED** [review]
Amazing work! 🥇 

Just left a small note about documenting the fertility quirk.

File: lib/accessory.js (hunk 1 above)
Comment (Kylmakalle) [reply lib/accessory.js]: Done 👌

**Review by hannseman — APPROVED** [review]
Nice!

**Comment by kinoushe** [comment]
Great work! I've been running a separate bridge just for Mi Flora for a long time, so it's great to be able to combine them.

The only issue I'm having is with FakeGato not working with the Mi Flora accessories. I have it enabled for each accessory but no graph data is showing in Eve, even after a few hours and restarts. The Hygrothermograph accessories are still working correctly.

**Comment by hannseman** [comment]
Hi @kinoushe, I think that graphing is only supported for humidity and temperature in Eve. See https://github.com/honkmaster/homebridge-mi-flower-care/issues/5

Or do you mean that not even humidity and temperature is graphed for the Mi Flora devices?

**Comment by kinoushe** [comment]
Hi @hannseman, yes - I mean just the humidity and temperature, the real time information is present and correct, but it isn't downloading the graph. The "dot" just pulses in the top right bar in Eve but never updates the graphs for the Mi Flora items.

I've removed the previous Mi Flora bridge and accessories, so I don't think it's a name/device conflict from the previous setup.

**Comment by hannseman** [comment]
Right! Think I found the issue. For Mi Flora devices we keep track of moisture separate from the humidity emitted by the “old” sensors. In the call to add an entry to fake-gato we guard that humidity is set before adding it:

  https://github.com/hannseman/homebridge-mi-hygrothermograph/blob/f6c9ff5db738c63757b979512f49e24c2e233821/lib/accessory.js#L418
  
I’ll push a fix in the weekend, thanks for the report 👍

**Comment by kinoushe** [comment]
Oh excellent, thanks @hannseman! I'll report back when the fix is out.

**Comment by hannseman** [comment]
@kinoushe just pushed 3.3.1 to npm with a possible fix.

**Comment by kinoushe** [comment]
The Eve graphs are now working correctly with this fix 👍 Thanks again @hannseman!
